static_frame/core/util.py

```diff
@@ -622,6 +649,60 @@ class IterNodeType(Enum):
     ITEMS = 2
 
 
+class SortStatus(Enum):
+    ASC = 1
+    DESC = 2
+    UNKNOWN = 3
+
+    def derive_status_from_slice(self, sl: slice) -> SortStatus:
+        if self is SortStatus.UNKNOWN:
+            return self
+
+        if sl.step is None or sl.step >= 1:
+            return self
+
+        # Reverse!
+        return SortStatus.DESC if self is SortStatus.ASC else SortStatus.ASC
+
+    @classmethod
+    def from_bool(cls, ascending: bool) -> tp.Literal[SortStatus.ASC, SortStatus.DESC]:
+        """
+        Derive the appropriate enum for the ascending argument.
+
+        If ascending is True, return `SortStatus.ASC`
+        If ascending is False, return `SortStatus.DESC`
+        """
+        return cls.ASC if ascending else cls.DESC
+
+    @classmethod
+    def from_sort_kwargs(
+        cls,
+        ascending: TBoolOrBools,
+        key: tp.Any,
+        kind: TSortKinds,
+    ) -> SortStatus:
+        """
+        Derive the appropriate enum for the ascending argument.
+
+        If ascending is a sequence of bools:
+        - `SortStatus.ASC` iif all flags are True
+        - `SortStatus.DESC` iif all flags are False
+        - `SortStatus.UNKNOWN` else
+
+        If key is not None, return `SortStatus.UNKNOWN`
+        If kind is not 'mergsort', return `SortStatus.UNKNOWN`
+        """
+        if key is not None or kind != DEFAULT_SORT_KIND:
+            return cls.UNKNOWN
+
+        if not isinstance(ascending, bool):
```

Comment:
I am not sure you should do this, but I had to test it:
```
>>> timeit isinstance(True, bool)
9.49 ns ± 0.148 ns per loop (mean ± std. dev. of 7 runs, 100,000,000 loops each)
>>> timeit True is True or True is False
7.46 ns ± 0.5 ns per loop (mean ± std. dev. of 7 runs, 100,000,000 loops each)
```

Reply:
Haha I like it